src/DataProtection/DataProtection/src/Repositories/RegistryXmlRepository.cs

```diff
@@ -155,4 +155,63 @@ private void StoreElementCore(XElement element, string valueName)
         // but the window for that should be small enough that we shouldn't have to worry about it.
         RegistryKey.SetValue(valueName, element.ToString(), RegistryValueKind.String);
     }
+
+    /// <inheritdoc/>
+    public virtual bool CanRemoveElements => true;
+
+    /// <inheritdoc/>
+    public virtual bool RemoveElements(Action<IReadOnlyCollection<IDeletableElement>> chooseElements)
+    {
+        ArgumentNullThrowHelper.ThrowIfNull(chooseElements);
+
+        var deletableElements = new List<DeletableElement>();
+
+        foreach (var valueName in RegistryKey.GetValueNames())
+        {
+            var element = ReadElementFromRegKey(RegistryKey, valueName);
+            if (element is not null)
+            {
+                deletableElements.Add(new DeletableElement(valueName, element));
+            }
+        }
+
+        chooseElements(deletableElements);
+
+        var allSucceeded = true;
+
+        foreach (var deletableElement in deletableElements)
+        {
+            if (deletableElement.ShouldDelete)
+            {
+                var valueName = deletableElement.ValueName;
+                _logger.RemovingDataFromRegistryKeyValue(RegistryKey, valueName);
+                try
+                {
+                    RegistryKey.DeleteValue(valueName);
+                }
+                catch (Exception ex)
+                {
+                    _logger.FailedToRemoveDataFromRegistryKeyValue(RegistryKey, valueName, ex);
+                    allSucceeded = false;
+                }
+            }
+        }
+
+        return allSucceeded;
+    }
+
+    private sealed class DeletableElement : IDeletableElement
```

Comment:
I believe it's `Deleteable` not `Deletable` ?

Reply:
I believe both are acceptable but no-e is more common, at least in the US.